CONTRIBUTING.md

```diff
@@ -34,6 +34,41 @@ Additionally, it may be necessary to determine build failures, these jobs are lo
 If a package continuously fails to build, the ROS Boss for that distribution may choose to revert the pull request that introduced it.
 Or they may take [other actions](https://github.com/ros-infrastructure/ros_buildfarm/blob/master/doc/ongoing_operations.rst) to avoid repeated failures.
 
+### Releasing third party system packages into a ROS distribution
+
+In general you should not override or replace system dependencies from the system distribution with a package in the ROS distribution.
+This is part of the software engineering process of maintaining the distribution that all developers can rely on specific versions of software to be available and remain consistent within the distribution.
+If you need newer versions of underlying libraries the you should either target newer versions with the minimum version available or if that’s not available help and encourage the upstream process to move forward to at least your minimum version for the next release of the distribution.
+
+Obviously this is a slow process and doesn’t fix things immediately for a short term.
+There are workarounds such as embedding copies of the libraries into your package or making another package with the newer library version.
+This usually can be made to work well in your local development environment.
+However if you want to release this style of workarounds you must make sure not to break other users who are expecting the system version either in downstream packages or elsewhere on the system.
+
+To release a ROS package which will overlay a system dependency it must:
+
+* Install side by side with the system one, with a different name.
+* Not cause compile errors for any packages including downstream ones
+* Not cause linking errors or crashes at runtime for yours or any downstream packages, including ones that use the system version on any of our target platforms.
+
+The release pipeline will only catch the first issue automatically.
+Manual review is required for the latter two.
+But in general if you want to do this without being disruptive you need to adjust paths to avoid installation collisions.
+Change all namespaces to avoid symbol collisions. And change header paths to not collide with the system ones if you export the include directories.
```

Comment:
```suggestion
But in general if you want to do this without being disruptive you need to:
1. Adjust paths to avoid installation collisions.
2. Change all namespaces to avoid symbol collisions
3. Change header paths to not collide with the system ones if you export the include directories
```